### PR TITLE
Convert Pet db to Libconfig

### PR DESCRIPTION
**Pull request by Asheraf**
[//]: # (**********************************)
[//]: # (** Fill in the following fields **)
[//]: # (**********************************)

[//]: # (Note: Lines beginning with syntax such as this one, are comments and will not be visible in your report!)

### Pull Request Prelude

[//]: # (Thank you for working on improving Hercules!)

[//]: # (Please complete these steps and check the following boxes by putting an `x` inside the brackets _before_ filing your Pull Request.)

- [x] I have followed [proper Hercules code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR will be closed if the above-mentioned criteria are not fulfilled.

### Known Issues and TODO List
- [x] test this stuff

[//]: # (Insert checklist here)
[//]: # (Syntax: - [ ] Checkbox)

[//]: # (**NOTE** Enable the setting "[√] Allow edits from maintainers." when creating your pull request if you have not already enabled it.)

[cont]: https://github.com/HerculesWS/Hercules/blob/master/CONTRIBUTING.md
[code]: https://github.com/HerculesWS/Hercules/wiki/Coding-Style

### CODE REVIEW

**Comment by HerculesWSAPI** [comment]
This change is [Reviewable](https://reviewable.io/reviews/herculesws/hercules/2000)

File: src/map/pet.c [hunk 1]
```diff
@@ -1199,145 +1200,210 @@ int pet_skill_support_timer(int tid, int64 tick, int id, intptr_t data) {
...
+	if (libconfig->setting_lookup_string(it, "EquipScript", &str))
+		pet->db[n].equip_script = *str ? script->parse(str, source, -pet->db[n].class_, SCRIPT_IGNORE_EXTERNAL_BRACKETS, NULL) : NULL;
+
+	return pet->db[n].class_;
+}
+
+bool pet_read_db_sub_intimacy(int idx, struct config_setting_t *t)
+{
+	int i32 = 0;
+
```
Comment (4144) [inline src/map/pet.c]: need assert for idx

File: src/map/pet.c [hunk 2]
```diff
@@ -1199,145 +1200,210 @@ int pet_skill_support_timer(int tid, int64 tick, int id, intptr_t data) {
...
-			if (*p != '{') {
-				ShowError("read_petdb: Invalid format (Pet Script column) in line %d, skipping.\n", lines);
-				continue;
-			}
+int pet_read_db_sub(struct config_setting_t *it, int n, const char *source)
+{
+	struct config_setting_t *t = NULL;
+	struct item_data *data = NULL;
+	const char *str = NULL;
+	int i32 = 0;
```
Comment (4144) [inline src/map/pet.c]: need assert for n

**Comment by idamonli** [comment]
You can implement pet evolution?

File: db/pet_db2.conf [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+//= GNU General Public License for more details.
+//=
+//= You should have received a copy of the GNU General Public License
+//= along with this program.  If not, see <http://www.gnu.org/licenses/>.
+//=========================================================================
+//= Pets Database
+//=========================================================================
+
+pet_db:(
+	// entries in this file will override the ones in /(pre-)re/pet_db.conf
```
Comment (MishimaHaruna) [inline db/pet_db2.conf]: Can you add the comment with the documentation about the existing keys and their meaning? (the same as in the other two .conf files)

File: db/pre-re/pet_db.conf [hunk 4]
```diff
@@ -0,0 +1,1544 @@
...
+	CaptureRate: capture rate            (int, defaults to 0)
+	Speed: speed                         (int, defaults to 0)
+	SpecialPerformance: true/false       (boolean, defaults to false)
+	TalkConvertClass: convert talk       (int, defaults to 0)
+	AttackRate: attack rate              (int, defaults to 0)
+	DefenceAttackRate: Defence attack    (int, defaults to 0)
+	ChangeTargetRate: change target      (int, defaults to 0)
+	PetScript: <" Pet Script (can also be multi-line) ">
+	EquipScript: <" Equip Script (can also be multi-line) ">
+},
```
Comment (MishimaHaruna) [inline db/pre-re/pet_db.conf]: I'd like to use this chance to change the spelling of some of these keys (because it gets more annoying to do this later, once people have converted their databases). My suggested keys are:

- `Id`: Ok.
- `SpriteName`: Ok
- `Name`: Ok
- `TamingItemID`: I'd shorten this to `TamingItem` to hint that we don't ask for a number
- `EggItemID`: As above, `EggItem`
- `AccessoryItemID`: As above, `AccessoryItem`
- `FoodItemID`: As above, `FoodItem`
- `Fullness`: I'd change this to the more meaningful `FoodEffectiveness`
- `HungryDelay`: I'd correct this to `HungerDelay`
- `Intimacy/Start`: I'd perhaps change this to `Intimacy/Initial`, or `Intimacy/Starting` if you prefer
- `Intimacy/Feed`: I'd clarify this as `Intimacy/FeedIncrement`
- `Intimacy/OverFeed`: Same as above, `Intimacy/OverFeedDecrement`
- `Intimacy/OwnerDie`: Same as above, `Intimacy/OwnerDeathDecrement`
- `CaptureRate`: Ok
- `Speed`: Ok
- `SpecialPerformance`: Ok
- `TalkConvertClass`: This should probably be a boolean, named `TalkWithEmotes` or `UseEmotes`
- `AttackRate`: Ok
- `DefenceAttackRate`: Should be corrected to `DefendRate` (considering "attack" above as a verb)
- `ChangeTargetRate`: Ok
- `PetScript`: Ok
- `EquipScript`: Ok
